doc/lsst.ip.diffim/AL_implementation.rst

```diff
@@ -0,0 +1,73 @@
+################################
+Alard-Lupton implementation code
+################################
+
+This page extends the API documentation of the implementation code.
+
+Solving for the matching kernel starts with the generation of the
+basis functions. The number of basis functions and the properties of
+the Gaussian bases (e.g. widths in pixels) are determined in
+`lsst.ip.diffim.makeKernelBasisList.generateAlardLuptonBasisList`. The
+decision logic depends on configuration and psf fwhm values of the
+images, some details is given in the function API documentation.
+
+:numref:`figclass` shows the main functional relation among the implementation
+classes. This diagram does not detail the abstraction hierarchy of the
+classes, they all have more general base classes. Please refer to the
+API documentation for details. There are also additional enums and
+classes declared to represent solution statuses and to select and
+configure numerical algorithms.  ``KernelPca`` and
+``KernelPcaVisitor`` are also not detailed here.
+
+.. _figclass:
+
+.. figure:: figures/AL_kernel_solution_classes.svg
+    :align: center
+    :alt: Kernel solution classes
+
+    Alard-Lupton kernel solution classes and their relationships.
+
+    Arrows point from the class that performs the arrow action on the
+    _pointed_ class. All containment or storing relations mean holding
+    of *shared_ptrs* of the object instances, there is no exclusive
+    ownership.
+
+..
+   Figure source on github:  
+   lsst-dm/diffimTests/tickets/DM-19443_visualize_coefficients/AL_kernel_solution_classes.drawio
+
+The solution is implemented in `lsst.ip.diffim.PsfMatchTask._solve`
+with an iterative loop to select good kernel solutions. In the C++ codebase, the
+solution uses the general visitor concept of ``SpatialCellSet`` to
+obtain local solutions across all over the image.
```

Comment:
This paragraph is closest to the gist I am seeking for a figure caption above, but I am not entirely sure what it is saying. Saying the solution is implemented [...] to select good solutions sounds circular. Please rephrase "across all over the image" to be less awkward.